csharp/src/Google.Protobuf.Test.NativeAot/Reflection/Extensions.cs

```diff
@@ -0,0 +1,185 @@
+            Assert.AreEqual(expectedMessage, ex.Message);
+        }
+
+        public static void GetExtensionEnumSingle()
+        {
+            var message = new TestAllExtensions();
+            message.SetExtension(OptionalForeignEnumExtension, ForeignEnum.ForeignBar);
+
+            var value = message.GetExtension(OptionalForeignEnumExtension);
+            Assert.AreEqual(ForeignEnum.ForeignBar, value);
+
+            Console.WriteLine($"Got value {value} for extension {OptionalForeignEnumExtension.FieldNumber}");
+        }
+
+        public static void GetExtensionRepeated()
+        {
+            var extensionValue = new TestAllTypes.Types.NestedMessage() { Bb = 42 };
+            var untypedExtension = new Extension<TestAllExtensions, IList>(RepeatedNestedMessageExtension.FieldNumber, codec: null);
+            var wrongTypedExtension = new RepeatedExtension<TestAllExtensions, TestAllTypes>(RepeatedNestedMessageExtension.FieldNumber, codec: null);
+
+            var message = new TestAllExtensions();
+
+            var value1 = message.GetExtension(untypedExtension);
+            Assert.IsNull(value1);
+
+            var repeatedField = message.GetOrInitializeExtension<TestAllTypes.Types.NestedMessage>(RepeatedNestedMessageExtension);
+            repeatedField.Add(extensionValue);
+
+            var value2 = message.GetExtension(untypedExtension);
+            Assert.IsNotNull(value2);
+            Assert.AreEqual(1, value2.Count);
+
+            var valueBytes = ((IMessage) value2[0]!).ToByteArray();
+            var parsedValue = TestProtos.Proto2.TestAllTypes.Types.NestedMessage.Parser.ParseFrom(valueBytes);
+            Assert.AreEqual(extensionValue, parsedValue);
+
+            var ex = Assert.Throws<InvalidOperationException>(() => message.GetExtension(wrongTypedExtension));
+
+            var expectedMessage = "The stored extension value has a type of 'Google.Protobuf.TestProtos.Proto2.TestAllTypes+Types+NestedMessage, Google.Protobuf.Test.TestProtos, Version=1.0.0.0, Culture=neutral, PublicKeyToken=a7d26565bac4d604'. " +
+                "This a different from the requested type of 'Google.Protobuf.TestProtos.Proto2.TestAllTypes, Google.Protobuf.Test.TestProtos, Version=1.0.0.0, Culture=neutral, PublicKeyToken=a7d26565bac4d604'.";
+            Assert.AreEqual(expectedMessage, ex.Message);
+        }
+
+        public static void GetExtensionValue()
+        {
+            var message = CreateFullTestAllExtensions();
+
+            Assert.AreEqual(true, message.GetExtension(OptionalBoolExtension));
+
+            var ex = Assert.Throws<NotSupportedException>(() => TestProtos.Proto2.TestAllExtensions.Descriptor.FindFieldByNumber(OptionalBoolExtension.FieldNumber).Accessor.GetValue(message));
+            Assert.AreEqual("Extensions reflection is not supported with AOT.", ex.Message);
+        }
+
+        public static void IsExtensionsInitialized()
```

Comment:
Rename to ExtensionsIsInitialized given that the method we're testing is IsInitialized?